Tools/clangd/clangd-config.yaml.jinja

```
@@ -0,0 +1,39 @@
+# !!! Autogenerated by generate-clangd-config !!!
+# ^ Removing this line will prevent this file from updating
+#
+# If you want to commit changes to the clangd config, edit
+# Tools/clangd/clangd-config.yaml.jinja instead.
+#
+# .clangd is a target generated during build.
+
+CompileFlags:
+    # clangd-16 doesn't support the -fcoroutines flag used by GCC.
+    Remove: [-fcoroutines]
```

When using GCC 13.2.0 and clang 18.1.3 I needed this:

```suggestion
    # clangd-16 doesn't support these flags used by GCC.
    Remove: [-fcoroutines, -Wno-stringop-overread, -Wno-stringop-overflow, -Wno-maybe-uninitialized]
```

---

I don't think its a blocker, but a nice improvement in the future would be to iterate over all flags and test with clang to get a more accurate list.

---

Added those flags.